src/axom/inlet/Table.hpp

```diff
@@ -371,63 +389,67 @@ class Table : public Verifiable
 
   /*!
    *****************************************************************************
-   * \brief Get an array represented as an unordered map from the input file
-   * of user-defined type
+   * \brief Add a dictionary of Integer Fields to the input file schema.
    *
-   * \param [out] map Unordered map to be populated with array contents
+   * \param [in] name Name of the dict
+   * \param [in] description Description of the dictionary
    *
-   * \return Whether or not the array was found
+   * \return Reference to the created dictionary
    *****************************************************************************
    */
-  template <typename T>
-  typename std::enable_if<!detail::is_inlet_primitive<T>::value, bool>::type
-  getArray(std::unordered_map<int, T>& map) const
-  {
-    if(m_sidreGroup->hasView("_inlet_array_indices"))
-    {
-      auto view = m_sidreGroup->getView("_inlet_array_indices");
-      int* array = view->getArray();
-      for(int i = 0; i < view->getNumElements(); i++)
-      {
-        auto index_label = std::to_string(array[i]);
-        map[array[i]] = getTable(index_label).get<T>();
-      }
-    }
-    else
-    {
-      SLIC_WARNING("[Inlet] Table does not contain an array");
-      return false;
-    }
-    return true;
-  }
+  Verifiable& addIntDict(const std::string& name,
+                         const std::string& description = "");
+  /*!
+   *****************************************************************************
+   * \brief Add a dictionary of Double Fields to the input file schema.
+   *
+   * \param [in] name Name of the dict
+   * \param [in] description Description of the dictionary
+   *
+   * \return Reference to the created dictionary
+   *****************************************************************************
+   */
+  Verifiable& addDoubleDict(const std::string& name,
+                            const std::string& description = "");
+
+  /*!
+   *****************************************************************************
+   * \brief Add a dictionary of String Fields to the input file schema.
+   *
+   * \param [in] name Name of the dict
+   * \param [in] description Description of the dictionary
+   *
+   * \return Reference to the created dictionary
+   *****************************************************************************
+   */
+  Verifiable& addStringDict(const std::string& name,
+                            const std::string& description = "");
+
+  /*!
+   *****************************************************************************
+   * \brief Add a dictionary of user-defined types to the input file schema.
+   *
+   * \param [in] name Name of the dict
+   * \param [in] description Description of the dictionary
+   *
+   * \return Reference to the created dictionary
+   *****************************************************************************
+   */
+  Table& addGenericDict(const std::string& name,
```

Comment:
Should the return type be a `Verifiable`? Or does it have to be a `Table`?

Reply:
Right now the `addGeneric*` functions return `Table` so the other `add*` functions are "accessible" - this prohibits an container of structs from being defined within a container of structs (which is detected and reported to the user when attempted).  Removing this limitation would require some modifications/extension of the `Verifiable` interface but is definitely possible.

Reply:
I was asking more from a consistency point of view -- i.e. will the return type make sense to users.

Reply:
My hope was that users wouldn't have to worry too much about the exact return type of these functions, but that the interface of the return type would reflect valid operations on the result. 

E.g., it wouldn't make sense to add a field to an array of primitives (as primitive arrays are "terminal"), but it would make sense to add a field to a struct (as structs are composite/product types).